bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitOHItemProxy.java

```diff
@@ -0,0 +1,165 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.io.homekit.internal;
+
+import static org.openhab.io.homekit.internal.HomekitCommandType.*;
+import static org.openhab.io.homekit.internal.HomekitDimmerMode.*;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.library.items.ColorItem;
+import org.eclipse.smarthome.core.library.items.DimmerItem;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Proxy class that can collect multiple commands for the same openHAB item and merge them to one command.
+ * e.g. Hue and Saturation update for Color Item
+ * 
+ * @author Eugen Freiter - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class HomekitOHItemProxy {
+    private final Logger logger = LoggerFactory.getLogger(HomekitOHItemProxy.class);
+
+    private static final int DEFAULT_DELAY = 50;
+
+    private final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
+
+    @Nullable
+    private ScheduledFuture<?> future;
+
+    private final Item item;
+
+    private HomekitDimmerMode dimmerMode = DIMMER_MODE_NORMAL;
+
+    // delay, how long wait for further commands. in ms.
+    private int delay = DEFAULT_DELAY;
+    private Map<HomekitCommandType, State> commandCache = new ConcurrentHashMap<>();
+
+    public HomekitOHItemProxy(final Item item) {
+        this.item = item;
+    }
+
+    public Item getItem() {
+        return item;
+    }
+
+    public void setDimmerMode(HomekitDimmerMode mode) {
+        dimmerMode = mode;
+    }
+
+    public void setDelay(int delay) {
+        this.delay = delay;
+    }
+
+    @SuppressWarnings("null")
+    private void sendCommand() {
+
+        if (!(item instanceof DimmerItem)) {
+            // currently supports only DimmerItem and ColorItem (which extends DimmerItem)
+            logger.debug("unexpected item type {}. Only DimmerItem and ColorItem are supported.", item);
+            return;
+        }
```

Comment:
Why not just make the `item` field a `DimmerItem`?

Reply:
i was just thinking, maybe in the future we will need to add further item types, e.g. groups.